### DISCUSSION TITLE
changing web app url without reloading or redirecting

### DISCUSSION THREAD

**Discussion by Pr47h4m** [csells/go_router#342] [Q&A] [answered]
Hello,
![UI with bottom navigation bar and routing](https://cdn.dribbble.com/users/2814348/screenshots/14896063/media/813f7a0aff3c6c55392520d2b91e9e34.png?compress=1&resize=1200x900&vertical=top)
I'm trying to implement this UI as a sample for **practicing go_router**
Here,
for changing views from `bottomNavigationBar` I'm changing _selectedIndex
so I specify view inside scaffold's body by `views[_selectedIndex]` (views is `List<Widget>`)
but **I also want url to get change** for example /home, /discover, /favourites, /profile but **I don't want to navigate** to other page since its a **bottom navigation bar**
So, if the web app is open UI can be controlled by updating `_selectedIndex` 
but when web app is opened from url then I can do something like `Home(viewIndex: 1)`, using `GoRoute`

### if not some updates in package can solve this problem and making go_router more usable

**Comment by csells** [comment, marked as answer]
Take a look at [the nested navigation sample](https://github.com/csells/go_router/blob/main/go_router/example/lib/nested_nav.dart) which does the same thing with a `TabView`.

**Reply by Pr47h4m** [reply]
Thanks for answer
